rspamd for aarch64 is broken on latest unstable

**Describe the bug**
rspamd for aarch64 can't find luajit and fails to build.

**To Reproduce**
`nix build github:NixOS/nixpkgs#legacyPackages.aarch64-linux.rspamd`

At the moment of writing, the latest unstable seems to be `34ad166a830d3ac1541dcce571c52231f2f0865a`.

**Expected behavior**
Build succeeds

**Screenshots**
n/a

**Additional context**
Hydra build log: https://hydra.nixos.org/build/128842971/nixlog/1

**Notify maintainers**
@avnik @fpletz @globin

**Metadata**
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - rspamd
# a list of nixos modules affected by the problem
module:
 - rspamd
```

I managed to pinpoint the issue to the commit `ff7792dd0568bec57c600bfbc7e1739d3d28e88b`, where rspamd was updated from 2.5 to 2.6.

Build log:
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/kgk44g6p7n44ck6v2yg2ylrf611miqiw-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/share/doc/rspamd -DCMAKE_INSTALL_INFODIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/include -DCMAKE_INSTALL_SBINDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_OSX_ARCHITECTURES=x86_64 -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/scxfwf6qnjc53vycr09rpziyjz3gajc9-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/scxfwf6qnjc53vycr09rpziyjz3gajc9-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/scxfwf6qnjc53vycr09rpziyjz3gajc9-binutils-2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/1pj0a1j37m6smzaj07sblnqwyg4skf3g-rspamd-2.6 -DDEBIAN_BUILD=ON -DRUNDIR=/run/rspamd -DDBDIR=/var/lib/rspamd -DLOGDIR=/var/log/rspamd -DLOCAL_CONFDIR=/etc/rspamd -DENABLE_JEMALLOC=ON
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /nix/store/qz3k9p38b8cjcan3i77aibv9gsm4bvcj-gcc-wrapper-9.3.0/bin/gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/qz3k9p38b8cjcan3i77aibv9gsm4bvcj-gcc-wrapper-9.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/qz3k9p38b8cjcan3i77aibv9gsm4bvcj-gcc-wrapper-9.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found RAGEL: /nix/store/vn8gfpilh6sm8l5z1xjki1kaxdpq7n07-ragel-6.10/bin/ragel (found version "6.10")
-- Found PkgConfig: pkg-config (found version "0.29.2")
-- Found Perl: /nix/store/z8bd9jdcw5vqhfbc9m9pdrkrwxl4swh7-perl-5.32.0/bin/perl (found version "5.32.0")
-- Using custom linker by name: gold
-- CMAKE_BUILD_TYPE: RELEASE
-- Enable IPO for the Release build
-- Final CFLAGS:  -O3 -DNDEBUG -O3 -funsafe-math-optimizations -fno-math-errno -fomit-frame-pointer
-- Final CXXFLAGS:  -O3 -DNDEBUG -O3 -funsafe-math-optimizations -fno-math-errno -fomit-frame-pointer
-- Compilation of slash macro convention asm set is -NOT- supported
-- Compilation of dollar macro convention asm set is -NOT- supported
-- Configuring for Linux
-- Checking for one of the modules 'luajit'
CMake Error at cmake/ProcessPackage.cmake:71 (MESSAGE):
  Cannot find library
  luajit;luajit-2.1;luajit2.1;luajit-2.0;luajit2.0;luajit-5.1 for package
  LIBLUAJIT
Call Stack (most recent call first):
  CMakeLists.txt:126 (ProcessPackage)


-- Configuring incomplete, errors occurred!
See also "/build/source/build/CMakeFiles/CMakeOutput.log".
```

It looks like rspamd is trying to find LuaJIT, but it's only available in `x86_64-linux` builds. Probable fix would be disabling LuaJIT support in CMake flags.

--- a/pkgs/servers/mail/rspamd/default.nix
+++ b/pkgs/servers/mail/rspamd/default.nix
@@ -33,7 +33,8 @@ stdenv.mkDerivation rec {
     "-DLOGDIR=/var/log/rspamd"
     "-DLOCAL_CONFDIR=/etc/rspamd"
     "-DENABLE_JEMALLOC=ON"
-  ] ++ lib.optional withHyperscan "-DENABLE_HYPERSCAN=ON";
+  ] ++ lib.optional withHyperscan "-DENABLE_HYPERSCAN=ON"
+  ++ lib.optional (!withLuaJIT) "-DENABLE_LUAJIT=OFF";
 
   passthru.tests.rspamd = nixosTests.rspamd;
 

